### ISSUE TITLE
Consideration for Excluding Item 14.5.6

### ISSUE THREAD

**Issue by ImanSharaf** [OWASP/ASVS#1793]
I'm writing to address a recent addition to the Application Security Verification Standard, specifically item 14.5.6, which states: "[ADDED] Verify that the infrastructure follows RFC 2616 and ignores the Content-Length header field if a Transfer-Encoding header field is also present."

My concern lies in the nature of item 14.5.6, which delves into compliance with RFC 2616 regarding HTTP header fields. This aspect seems more inherently related to infrastructure-level concerns rather than application-level security practices, which are the primary focus of ASVS.

**Comment by elarlang** [comment]
Quick answer - I think it belongs to the scope.

To not have another discussion in another issue about the scope, I'll explain my vision about it through other channels first.

**Comment by elarlang** [comment]
TLDR:
  * (in my opinion) requirement is in scope
  * requirement text should mention "HTTP Request Smuggling"
  * the word "infrastructure" is a bit too "wide" and should be finetuned in the requirement text.

The requirement got in via https://github.com/OWASP/ASVS/issues/1153 > https://github.com/OWASP/ASVS/pull/1155

The point for the requirement is to provide defense against HTTP Request Smuggling. It was mentioned in initial proposal but was removed later:
https://github.com/OWASP/ASVS/pull/1155/commits/282bbd2def087f159abeeb39c104970a81b7b9a6

Current requirement:
| # | Description | L1 | L2 | L3 | CWE |
| :---: | :--- | :---: | :---: | :---: | :---: |
| **14.5.6** | [ADDED] Verify that the infrastructure follows RFC 2616 and ignores the Content-Length header field if a Transfer-Encoding header field is also present. | | ✓ | ✓ | 444 |

In the requirement context, the word "infrastructure" means load balancer, proxy, or web application firewall. I think it can be misinterpreted and should be finetuned.

In this context I consider load balancers, proxies, and web application firewalls to be in the ASVS scope, as they modify, validate, or filter HTTP request before it reaches the application server.

**Comment by elarlang** [comment]
Proposal:
_Verify that all application components, such as load balancer, firewall, and application server, follow RFC 2616 and ignore the Content-Length header field if a Transfer-Encoding header field is also present to prevent HTTP Request Smuggling._

ping @ImanSharaf @tghosth @jmanico

**Comment by jmanico** [comment]
I like this. How about:

_Verify that all application components, including load balancers, firewalls, and application servers, comply with RFC 2616 by ignoring the Content-Length header when a Transfer-Encoding header is present, to prevent HTTP Request Smuggling._
